src/controllers/keyboard/keyboardshortcutsupdater.cpp

```diff
@@ -0,0 +1,79 @@
+#include <QAction>
+
+#include "controllers/keyboard/keyboardshortcutsupdater.h"
+
+KeyboardShortcutsUpdater::KeyboardShortcutsUpdater() { }
+KeyboardShortcutsUpdater::~KeyboardShortcutsUpdater() { }
+
+void KeyboardShortcutsUpdater::addWatcher(ShortcutChangeWatcher* watcher) {
+    // TODO(Tomasito) Add test to see if there is already a watcher with the same ConfigKey. If there is,
+    // ...            replace the watcher with the new one. (maybe QMultiHash<ConfigKey, ShortcutChangeWatcher> ?)
+
+    m_shortcutChangeWatchers.append(watcher);
+}
+
+// NOTE(Tomasito): There shouldn't be more than one watchers with the same ConfigKey, but if
+// ...             there are, this will return the first one that is found.
+ShortcutChangeWatcher* KeyboardShortcutsUpdater::getWatcher(ConfigKey configKey) {
+    for (ShortcutChangeWatcher* watcher: m_shortcutChangeWatchers) {
+        if (watcher->m_configKey == configKey) {
+            return watcher;
+        }
+    }
+    return nullptr;
+}
+
+void KeyboardShortcutsUpdater::slotUpdateShortcuts(KeyboardControllerPresetPointer pKbdPreset) {
+    QMultiHash<QString, ConfigKey> keyboardShortcuts =
+            pKbdPreset->getMappingByGroup("[KeyboardShortcuts]");
+
+    for (ShortcutChangeWatcher* watcher: m_shortcutChangeWatchers) {
+        watcher->updateShortcut(&keyboardShortcuts);
+    }
+}
+
+
+
+ShortcutChangeWatcher::ShortcutChangeWatcher(QAction* action, ConfigKey configKey, QKeySequence defaultKeySeq) :
+        QObject(action),
+        m_configKey(configKey),
+        m_pAction(action),
+        m_defaultKeySeq(defaultKeySeq) {
+
+    restoreDefault();
+}
+
+ShortcutChangeWatcher::~ShortcutChangeWatcher() { }
+
+void ShortcutChangeWatcher::updateShortcut(QMultiHash<QString, ConfigKey>* pShortcuts) {
+    // Check if shortcut is found in the given QMultiHash
+    bool shortcutFound = false;
```

Comment:
I didn't return from the body of the loop because I wanted to make sure that the last found one was returned, not the first (if multiple found with same key). However, QMultiHash being a Hash, order is arbitrary, and thus it made not much sense :) It's changed!